java/api-test/pom.xml

```diff
@@ -42,4 +37,37 @@
         </dependencies>
     </dependencyManagement>
 
+    <profiles>
+         <profile>
+            <id>compile-jdk8</id>
+            <properties>
+                <jdk.version>1.8</jdk.version>
+                <maven.compiler.source>1.8</maven.compiler.source>
+                <maven.compiler.target>1.8</maven.compiler.target>
+                <maven.compiler.testSource>1.8</maven.compiler.testSource>
+                <maven.compiler.testTarget>1.8</maven.compiler.testTarget>
+            </properties>
+        </profile>
+        <profile>
+            <id>compile-jdk11</id>
+            <properties>
+                <jdk.version>11</jdk.version>
+                <maven.compiler.source>11</maven.compiler.source>
```

Comment:
<maven.compiler.source> should always be 1.8. We want to make sure no Java API creeps in that is not compatible with 1.8. The only exception to this is the samples for which <maven.compiler.source> should be 17. 

Also, it seems to me that these profiles should be in java/pom.xml and would not be needed here. But, maybe I'm missing something.

Reply:
I set compiler.source to 1.8 and compiler.testSource to 1.8 on all profiles. Does that sounds right or should I delete the profiles and just set the compiler.source to 1.8 as a global property?